### ISSUE TITLE
Multiple header lines

### ISSUE THREAD

**Issue by lafather** [JoshClose/CsvHelper#746]
Hello!

Is there any possibility to write headers to the same CSV file multiple times? I am trying to write records of different classes and would like each of them to have its own headers line.

Best regards,
Vitalii

**Comment by JoshClose** [comment]
In the 3.0 pre-release you can. You can do it as many times as you want on as many rows as you want. This will RTM in the next few weeks.

In the current version, you can just write strings every time you want to write headers.

**Comment by lafather** [comment]
Is pre-release version compatible with .NET Framework 4.5.1?

**Comment by JoshClose** [comment]
Currently, no. Only 4.6.1. https://docs.microsoft.com/en-us/dotnet/standard/net-standard

What is your reason for using 4.5.1 instead of 4.6.1? I'd like to only publish a netstandard2.0 version and anyone that wants to use previous versions of .net can use the csvhelper 2.x release, but I'm not completely set on this.

**Comment by lafather** [comment]
No reason, I'm working on a legacy project which targets 4.5.1. That's why I asked this question.
Thanks for your answers.

**Comment by JoshClose** [comment]
In 2.x, you can do something like this. I would suggest real types with a class map though.

```cs
void Main()
{
	using (var stream = new MemoryStream())
	using (var writer = new StreamWriter(stream))
	using (var reader = new StreamReader(stream))
	using (var csv = new CsvWriter(writer))
	{
		var records1 = new List<object>
		{
			new { A = 1, B = "one" },
			new { A = 2, B = "two" },
		};

		var records2 = new List<object>
		{
			new { C = 3, D = "three" },
			new { C = 4, D = "four" },
		};
		
		csv.Configuration.HasHeaderRecord = false;
		
		csv.WriteField("A");
		csv.WriteField("B");
		csv.NextRecord();
		csv.WriteRecords(records1);
		
		csv.NextRecord();
		
		csv.WriteField("C");		
		csv.WriteField("D");
		csv.NextRecord();
		csv.WriteRecords(records2);
		
		writer.Flush();
		stream.Position = 0;
		
		reader.ReadToEnd().Dump();
	}
}
```

**Comment by tek-andj** [comment]
Hi Josh,

Could you please give an example on how to do this in 3.0 prerelease ? Trying with RC5, and still getting the "header has already been written" error.

**Comment by JoshClose** [comment]
I ran that code as is in 3.0.0 and it worked fine. 3.0.0 has been released!

**Comment by tek-andj** [comment]
Creating a csvwriter, and calling
WriteHeader<T>
NextRecord
WriteHeader<V>

Gives the error that header is already written.
Could you give a working example?

Den 9. okt. 2017 kl. 17.44 skrev Josh Close <notifications@github.com<mailto:notifications@github.com>>:


I ran that code as is in 3.0.0 and it worked fine. 3.0.0 has been released!

-

**Comment by JoshClose** [comment]
Yeah, that doesn't work. I don't know what's going on. I'll fix it. Maybe it was in a branch and didn't get merged in or something, because I'm positive I did that.

**Comment by JoshClose** [comment]
@lafather 3.0.0 has a .NET 4.5 build in it

**Comment by JoshClose** [comment]
@tek-andj I fixed this and it will be in 3.0.1 sometime today.

**Comment by tek-andj** [comment]
Thank you very much :)
